### PR TITLE
kubetest/kind: apply more bug fixes

### PR DESCRIPTION
**Pull request by neolit123**
had to run this locally to be able to catch the rest of the problems.

- Remove warning in setKubeConfigEnv
control.FinishRunning() prints an error for that already.
- Update Up()
  - comment on top of function
  - use an arg list, as passing "" in case of missing config
    or image flags throws errors
- make Down() non-fatal. comment in the diff explains why.
- make KubectlCommand() return just the command without setting
  environment. Up() and TestSetup() already do that.

seems to work now, using `--up --test --down`.

/kind bug
/assign @BenTheElder

### CODE REVIEW

File: kubetest/kind/kind.go [hunk 1]
```diff
@@ -350,9 +345,9 @@ func (d *Deployer) TestSetup() error {
 // Down tears down the cluster.
 func (d *Deployer) Down() error {
 	cmd := exec.Command("kind", "delete", "cluster")
-	if err := d.control.FinishRunning(cmd); err != nil {
-		return err
-	}
+	// This function seems to be called even for --up only,
```
Comment (BenTheElder) [inline kubetest/kind/kind.go]: @krzyzacy ??

Comment (krzyzacy) [reply kubetest/kind/kind.go]: For the case here, check `IsUp()` first?

Comment (BenTheElder) [reply kubetest/kind/kind.go]: clarification from sen: `--up` will try to tear down the previous cluster, so we should make sure IsUp is respected

Comment (BenTheElder) [reply kubetest/kind/kind.go]: we can revisit this later

**Pull request merged**